Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -4394,6 +4465,357 @@ def arg_to_datetime(arg, arg_name=None, is_utc=True, required=False, settings=No
+                       'sent-by': 'sent',
+                       'sent-from': 'received-by',
+                       'sent-to': 'received-by',
+                       'similar-to': 'similar-to',
+                       'sub-domain-of': 'supra-domain-of',
+                       'supra-domain-of': 'sub-domain-of',
+                       'targeted-by': 'targets',
+                       'targets': 'targeted-by',
+                       'Types': 'Reverse',
+                       'uploaded-to': 'hosts',
+                       'used-by': 'uses',
+                       'used-on': 'targeted-by',
+                       'uses': 'used-by',
+                       'variant-of': 'variant-of'}
+
+    @staticmethod
+    def is_valid(_type):
+        # type: (str) -> bool
+
+        return _type in Relations.RELATIONS_NAMES.keys()
+
+    @staticmethod
+    def get_reverse(name):
+        # type: (str) -> str
+
+        return Relations.RELATIONS_NAMES[name]
+
+
+class EntityRelation:
+    """
+    XSOAR entity relation.
+
+    :type name: ``str``
+    :param name: Relation name.
+
+    :type relation_type: ``str``
+    :param relation_type: Relation type.
+
+    :type entity_a: ``str``
+    :param entity_a: A value, A aka Source of the relation.
+
+    :type entity_a_family: ``str``
+    :param entity_a_family: Entity family of type A, A aka Source of the relation. (e.g. IP/URL/...).
+
+    :type object_type_a: ``str``
+    :param object_type_a: Entity type B, B aka Source of the relation. (For future use).
+
+    :type entity_b: ``str``
+    :param entity_b: B value, B aka Source of the relation.
+
+    :type entity_b_family: ``str``
+    :param entity_b_family: Entity family of type B, B aka Source of the relation. (e.g. IP/URL/...)
+
+    :type object_type_b: ``str``
+    :param object_type_b: Entity type B, B aka Source of the relation. (For future use).
+
+    :type source_reliability: ``str``
+    :param source_reliability: Source_reliability.
+
+    :type fields: ``dict``
+    :param fields: Custom fields.
+
+    :type brand: ``str``
+    :param brand: Source brand name. (Optional)
+
+    :return: None
+    :rtype: ``None``
+    """
+
+    def __init__(self, name, entity_a, object_type_a, entity_b, object_type_b,
+                 relation_type='indicatorToIndicator', entity_a_family='Indicator', entity_b_family='Indicator',
+                 source_reliability="", fields=None, brand=""):
+
+        # Relation
+        if not Relations.is_valid(name):
+            raise ValueError("Invalid relation: " + name)
+        self._name = str(name)
+
+        self._reverse_name = Relations.get_reverse(str(name))
+
+        if not RelationsTypes.is_valid_type(relation_type):
+            raise ValueError("Invalid relation type: " + relation_type)
+        self._relation_type = str(relation_type)
+
+        # Entity A - Source
+        self._entity_a = str(entity_a)
+
+        # The type of the entity doesnt have to be a indicator type
```

Comment:
what else can it be?

Reply:
Currently only indicator types , removed the validation after the meeting.